### PR TITLE
Employees jion meeting

### PR DESCRIPTION
**Pull request by Graciexia**
add invitation and time overlap function

### CODE REVIEW

File: app/controllers/meetings_controller.rb [hunk 1]
```diff
@@ -91,18 +120,18 @@ def destroy
...
-  #     redirect_to @meeting, alert: 'You are already in another meeting.'
-  #   end
-  # end
-
-
+  # An employee cannot create or join a meeting that takes place during another meeting they own or are part of.
+  def meeting_overlap? (check_meeting)
+    all_meetings = Meeting.where(employee_id: current_employee.id) # owned meetings
+    all_meetings += EmployeeMeeting.where(employee_id: current_employee.id).map{|em| em.meeting} # attending meetings
+    all_meetings.each do |meeting|
```
Comment (JMD1986) [inline app/controllers/meetings_controller.rb:127]: :+1:

File: app/controllers/meetings_controller.rb [hunk 2]
```diff
@@ -91,18 +120,18 @@ def destroy
...
+      if (meeting.start_time <= check_meeting.start_time and check_meeting.start_time <= meeting.end_time) or
```
Comment (jah2488) [inline app/controllers/meetings_controller.rb]: We don't want to ever use `and` in ruby and instead use `&&`. http://devblog.avdi.org/2010/08/02/using-and-and-or-in-ruby/

**Comment by jah2488** [comment]
Remove `/.idea` in the next commit and change all occurrences of `and` to `&&` and we can merge.

**Comment by Graciexia** [comment]
Remove /.idea in the next commit and change all occurrences of and to &&

File: app/controllers/meetings_controller.rb [hunk 3]
```diff
@@ -91,18 +120,18 @@ def destroy
...
-  # end
-
-
+  # An employee cannot create or join a meeting that takes place during another meeting they own or are part of.
+  def meeting_overlap? (check_meeting)
+    all_meetings = Meeting.where(employee_id: current_employee.id) # owned meetings
+    all_meetings += EmployeeMeeting.where(employee_id: current_employee.id).map{|em| em.meeting} # attending meetings
+    all_meetings.each do |meeting|
+      if (meeting.start_time <= check_meeting.start_time && check_meeting.start_time <= meeting.end_time) ||
+          (meeting.start_time <= check_meeting.end_time && check_meeting.end_time <= meeting.end_time)
```
Comment (jah2488) [inline app/controllers/meetings_controller.rb:129]: :+1:

**Pull request merged**